src/main/java/org/truffleruby/core/exception/CoreExceptions.java

```diff
@@ -613,46 +640,18 @@ public DynamicObject noSuperMethodOutsideMethodError(Node currentNode) {
         DynamicObject noMethodError = Layouts.NAME_ERROR.createNameError(
                 context.getCoreLibrary().getNoMethodErrorFactory(),
                 messageString,
+                null,
                 backtrace,
                 null,
                 context.getSymbolTable().getSymbol("<unknown>"));
         // FIXME: the name of the method is not known in this case currently
         return noMethodError;
     }
 
-    @TruffleBoundary
-    public DynamicObject noSuperMethodError(String name, Object self, Object[] args,  Node currentNode) {
-        return noMethodError(StringUtils.format("super: no superclass method `%s'", name), self, name, args, currentNode);
-    }
-
-    @TruffleBoundary
-    public DynamicObject noMethodErrorOnReceiver(String name, Object receiver, Object[] args, Node currentNode) {
-        final DynamicObject logicalClass = context.getCoreLibrary().getLogicalClass(receiver);
-        final String moduleName = Layouts.MODULE.getFields(logicalClass).getName();
-
-        // e.g. BasicObject does not have inspect
-        final boolean hasInspect = ModuleOperations.lookupMethod(logicalClass, "inspect", Visibility.PUBLIC) != null;
-        final Object stringRepresentation = hasInspect ? context.send(receiver, "inspect", null) : context.getCoreLibrary().getNil();
-
-        return noMethodError(StringUtils.format("undefined method `%s' for %s:%s", name, stringRepresentation, moduleName), receiver, name, args, currentNode);
-    }
```

Comment:
Please rebase since d891d0e6c96f8ba17b0eba763cc93c7d5b5077b9 was merged and used this method.